src/scale.ts

```diff
@@ -53,12 +54,18 @@ const CONTINUOUS_DOMAIN_INDEX = toSet(CONTINUOUS_DOMAIN_SCALES);
 export const DISCRETE_DOMAIN_SCALES: ScaleType[] = ['ordinal', 'point', 'band'];
 const DISCRETE_DOMAIN_INDEX = toSet(DISCRETE_DOMAIN_SCALES);
 
+const BIN_SCALES_INDEX = toSet(['bin-linear', 'bin-ordinal']);
+
 export const TIME_SCALE_TYPES: ScaleType[] = ['time', 'utc'];
 
 export function hasDiscreteDomain(type: ScaleType): type is 'ordinal' | 'point' | 'band' {
   return type in DISCRETE_DOMAIN_INDEX;
 }
 
+export function isBinScale(type: ScaleType): type is 'bin-linear' | 'bin-ordinal' {
```

Comment:
As mentioned on slack, it's ok to add `isBinScale`.  

However, you should also make `hasContinuousDomain` opposite of `hasDiscreteDomain()` because our logic are written with this assumption.  Otherwise, we will definitely get some other code wrong. 

bin-ordinal = discrete
bin-linear = continuous